bugfix: fix several minor issues

liushuyu opened the pull request:
- Fixed `-frust-crate= option` got incorrectly overridden by a default value (`example`)
- Fix a minor typo in `gcc/rust/ast/rust-ast-full-test.cc`

CohenArthur reviewed (approved):
Thanks! I guess this ties in to [this issue](https://github.com/Rust-GCC/gccrs/issues/789) which is also related to the crate name. LGTM :)

liushuyu commented:
> Thanks! I guess this ties in to [this issue](https://github.com/Rust-GCC/gccrs/issues/789) which is also related to the crate name. LGTM :)

You are right! I guess the default crate name should be the file stem of the source file name.

philberty commented:
Great work @liushuyu i think we just need you to remove auto generated stuff in the PR comment. "Thank you for making Rust GCC better!" otherwise this will end up in the git history we just need the comment relating to these changes. Then we can merge this.

liushuyu commented:
> Great work @liushuyu i think we just need you to remove auto generated stuff in the PR comment. "Thank you for making Rust GCC better!" otherwise this will end up in the git history we just need the comment relating to these changes. Then we can merge this.

Oh, I did not know that. I have removed the extra contents now, and for the record, I will move the note I added to this comment.

---

### Considerations

- In `rustc`, when the crate name is not supplied, `rustc` will substitute the name `main` into the crate name when emitting symbols (observed behavior, I have yet to find where this logic is in `rustc`).
